v2/AdminCore.WebApi.Tests/Controllers/DashboardControllerTest.cs

```diff
@@ -0,0 +1,167 @@
+﻿using AdminCore.Common.Interfaces;
+using AdminCore.DTOs.Dashboard;
+using AdminCore.DTOs.Event;
+using AdminCore.WebApi.Controllers;
+using AdminCore.WebApi.Models.Dashboard;
+using AutoFixture;
+using AutoMapper;
+using NSubstitute;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace AdminCore.WebApi.Tests.Controllers
+{
+  public class DashboardControllerTest : BaseControllerTest
+  {
+    private readonly DashboardController _dashboardController;
+    private readonly IDashboardService _dashboardService;
+    private readonly IEmployeeCredentials _employeeCredentials;
+    private readonly Fixture _fixture;
+    private readonly IMapper _mapper;
+
+    public DashboardControllerTest()
+    {
+      _fixture = new Fixture();
+
+      _dashboardService = Substitute.For<IDashboardService>();
+      _employeeCredentials = Substitute.For<IEmployeeCredentials>();
+      _mapper = Substitute.For<IMapper>();
+
+      _dashboardController = new DashboardController(_dashboardService, _employeeCredentials, _mapper);
+    }
+
+    [Fact]
+    public void GetDashboardSnapshot_WhenCalled_ReturnsDashboardSnapshot()
+    {
+      // Arrange
+      var numberOfSnapshotModels = 4;
+
+      var snapshotModel = _fixture.CreateMany<TeamSnapshotViewModel>(numberOfSnapshotModels);
+      var snapshotReturnedFromService = _fixture.Create<Dictionary<string, List<EmployeeSnapshotDto>>>();
+
+      _dashboardService.GetTeamSnapshotDashboardEvents().Returns(snapshotReturnedFromService);
+
+      _mapper.Map<Dictionary<string, List<EmployeeSnapshotDto>>, List<TeamSnapshotViewModel>>(snapshotReturnedFromService).Returns(snapshotModel);
+
+      // Act
+      var result = _dashboardController.GetDashboardSnapshot();
+
+      // Assert
+      RetrieveValueFromResult<List<TeamSnapshotViewModel>>(result);
+
+      _dashboardService.Received(1).GetTeamSnapshotDashboardEvents();
+
+      _mapper.Received(1).Map<Dictionary<string, List<EmployeeSnapshotDto>>, TeamSnapshotViewModel>(Arg.Is(snapshotReturnedFromService));      
+    }
+
+    [Fact]
+    public void GetEmployeeEvents_WhenCalled_ReturnsEmployeeEvents()
+    {
+      // Arrange
+      var employeeId = 88;
+      var numberOfEvents = 5;
+      var searchDate = DateTime.Now;
+
+      var eventsReturnedFromService = _fixture.CreateMany<EventDto>(numberOfEvents).ToList();
+      var eventModels = _fixture.CreateMany<DashboardEventViewModel>(numberOfEvents).ToList();
+
+      _dashboardService.GetEmployeeDashboardEvents(employeeId, searchDate).Returns(eventsReturnedFromService);
+
+      _mapper.Map<IList<EventDto>, List<DashboardEventViewModel>>(Arg.Is(eventsReturnedFromService)).Returns(eventModels);
+
+      _employeeCredentials.GetUserId().Returns(employeeId);
+
+      // Act
+      var result = _dashboardController.GetEmployeeEvents(searchDate);
+
+      // Assert
+      RetrieveValueFromResult<EmployeeEventViewModel>(result);
```

Comment:
Again as above

Reply:
Updated - Setting what's returned and when applicable checking the count is correct.